doc/releases/release-notes-3.6.rst

```diff
@@ -149,6 +149,16 @@ Boards & SoC Support
 Build system and infrastructure
 *******************************
 
+* Added functionality for Link Time Optimisation.
+  This change includes interrupt script generator rebuilding and adding following options:
+
+  - :kconfig:option:`CONFIG_ISR_TABLES_LOCAL_DECLARATION` Kconfig option:
+  LTO compatible interrupt tables parser,
+  - :kconfig:option:`CONFIG_LTO` Kconfig option: Enable Link Time Optimisation.
+
+  Currently the LTO compatible interrupt tables parser is only supported by ARM architectures and
+  GCC compiler/linker.
+
```

Comment:
It's probably worth adding a mention of the PR number here for anyone interested in the details (https://github.com/zephyrproject-rtos/zephyr/pull/66392)

Reply:
Done.